core/src/main/scala/org/bykn/bosatsu/Externals.scala

```diff
@@ -31,19 +33,16 @@ sealed abstract class FfiCall {
       val m = cls.getMethod(parts.last, args.init :_*)
       val inst = instFn(cls)
 
-      def invoke(tpe: rankn.Type, args: List[Any]): Any =
+      def invoke(tpe: rankn.Type, args: List[Value]): Value =
         tpe match {
           case rankn.Type.ForAll(_, t) => invoke(t, args)
           case rankn.Type.Fun(a, tail) =>
-            new Fn[Any, Any] {
-              val tpeStr = TypeRef.fromType(tpe).get.toDoc.render(80)
-              override def toString = s"$tpeStr, args: $args"
-              def apply(x: Any) = {
-                invoke(tail, x :: args)
-              }
+            Value.FnValue { x =>
+              Eval.always(invoke(tail, x :: args))
             }
           case _ =>
             m.invoke(inst, args.reverse.toArray.asInstanceOf[Array[AnyRef]]: _*)
+              .asInstanceOf[Value]
```

Comment:
maybe we should change the FFI to just require using `Evaluation.Value`, which removes these casts.